posthog/temporal/batch_exports/destination_tests.py

```diff
@@ -793,20 +793,117 @@ async def _run_step(self) -> DestinationTestStepResult:
             warehouse=self.warehouse,
         )
 
-        with connection.cursor() as cursor:
-            try:
+        with connection:
+            with connection.cursor() as cursor:
+                try:
+                    _ = cursor.execute(f'USE DATABASE "{self.database}"')
+                except ProgrammingError as err:
+                    if err.msg is not None and "Object does not exist" in err.msg:
+                        return DestinationTestStepResult(
+                            status=Status.FAILED,
+                            message=f"The configured database '{self.database}' does not exist or we are missing 'USAGE' permissions on it.",
+                        )
+                    else:
+                        return DestinationTestStepResult(
+                            status=Status.FAILED,
+                            message=f"Could not use Snowflake database '{self.database}'. Received error: {err}.",
+                        )
+
+        return DestinationTestStepResult(
+            status=Status.PASSED,
+        )
+
+
+class SnowflakeSchemaTestStep(DestinationTestStep):
```

Comment:
There's quite a bit of duplication in all the setup code. Don't necessarily need to do anything about it now but we could probably consolidate quite a bit of this in a base class.

Reply:
Ideally, some of this could be abstracted in the batch export itself, and then we just import it here. But otherwise yeah, in the base class.

Reply:
yeah, true, that would be even better :)